### ISSUE TITLE
[Bug] Unable to display the configuration screen

### ISSUE THREAD

**Issue by xingnengmao666** [BloCamLimb/ModernUI#240]
**Describe the bug**
Unable to display the configuration screen

**To Reproduce**
Open The configuration screen(Ctrl+K)

**Expected behavior**
Opens the configuration screen and displays the options

**Screenshots**
![image](https://github.com/user-attachments/assets/42f84882-5379-4508-ab5a-3839315b65da)

**Environment**
 - Mod Version:3.11.0.4
 - Forge Version: （Fabric）0.16
 - Operating System:Windows 11
 - Graphics Card:Iris Xe
 - Other Mods:(Too much)

**Comment by xingnengmao666** [comment]
https://mcrconsle-my.sharepoint.com/:u:/g/personal/admin_mcrconsle_onmicrosoft_com/EZs7qCY05RdBv1nxDCOOxu4B69-8P_sutadoWa2jwvvsjQ?e=gymUub    The mods

**Comment by BloCamLimb** [comment]
Paste your game log.

**Comment by xingnengmao666** [comment]
[latest.log](https://github.com/user-attachments/files/17009322/latest.log)
如果你可以说中文的话，可以使用中文交流，谢谢

**Comment by YIEMM** [comment]
I have the same problem

**Comment by YIEMM** [comment]
However, the version is 3.11.0.2
minecraft version is 1.20.4

**Comment by BloCamLimb** [comment]
这可以确定是GPU指定的问题。你需要开启ModernUI的开发人员模式，并在打开任意ModernUI界面后按Ctrl+Shift+Y保存截图，看看是否正常渲染。
找到config/ModernUI/common.toml将enableDeveloperMode设为true开启开发人员模式。

**Comment by xingnengmao666** [comment]
![image-2024_09_16_15_10_10](https://github.com/user-attachments/assets/346c4bde-7b90-4f99-9edc-fe1e746b3a4a)
看来并不能正确渲染orz

**Comment by xingnengmao666** [comment]
![image](https://github.com/user-attachments/assets/7532ab4f-7da8-40b3-ba33-a4a93bc08da4)
开发者模式开起了

**Comment by xingnengmao666** [comment]
值得一提的是，有些时候打开配置页面会出现几条闪烁的横线后快速消失

**Comment by BloCamLimb** [comment]
建议用RenderDoc抓帧测试，不会可以进群。993254209

**Comment by BloCamLimb** [comment]
> However, the version is 3.11.0.2 minecraft version is 1.20.4

It's known that there is rendering issue on Intel Iris Xe Graphics, but I don't have this GPU for testing. If you are experienced in GPU debugging, you could use RenderDoc to capture frames and share the results in my Discord server.

**Comment by xingnengmao666** [comment]
发申请了，头像是个二维码的那位

**Comment by xingnengmao666** [comment]
此问题非常魔幻的解决了……抓帧一开，ui就出来了……

**Comment by xingnengmao666** [comment]
https://www.bilibili.com/video/BV12e4y1D7oT 如果你也想这样玩但是不会用抓帧软件的话可以参照这个视频

**Comment by BloCamLimb** [comment]
I've found that disabling DSA can workaround the bug.

**Comment by YIEMM** [comment]
![image-2024_09_17_22_18_02](https://github.com/user-attachments/assets/d1f87e16-84b6-45f8-b24d-c691fd803b03)

**Comment by Brancieq-Paul** [comment]
Same problem, with a RTX 3060 Laptop, same version of the mod and ssame version of fabric

**Comment by BloCamLimb** [comment]
> Same problem, with a RTX 3060 Laptop, same version of the mod and ssame version of fabric

I see no reason for this issue to occur on NVIDIA GPUs. Since you have the laptop, make sure your NVIDIA driver is up-to-date and use the dedicated NVIDIA graphics processor to launch Minecraft. You can verify the GPU information in the game log and F3 debug screen.

**Comment by yuseiFudu** [comment]
> I've found that disabling DSA can workaround the bug.
What is the DSA? because im having the same bug with 1.20.6

**Comment by BloCamLimb** [comment]
The term DSA refers to OpenGL Direct State Access, it is a technology that drivers may or may not support. When the driver reports that DSA is supported, ModernUI will call DSA-like GL functions. But Intel's drivers have bugs to support it, then ModernUI can apply a workaround, disable DSA for Intel GPUs. ModernUI 3.11 does not provide any method for users to manually disable it, so you will need to wait for the next update. You can report this to Intel engineers, but I don't think they will fix it in the short term.

The specific OpenGL functions related to the driver bug are: glCreateVertexArrays, glVertexArrayAttribFormat, glVertexArrayAttribBinding

The interesting thing is that when you start Minecraft with RenderDoc, there won't be any issues.

> > I've found that disabling DSA can workaround the bug.
> > What is the DSA? because im having the same bug with 1.20.6

**Comment by lassuly** [comment]
Yes, my situation is exactly the same as his. This problem has been bothering me for a long time and I hope the developers will update it as soon as possible.
